### PR TITLE
[WIP] ideas for generating docker images

### PR DESCRIPTION
**Pull request by kyclark**
## Description

These are just basic sketches for a possible improvement over "docker.sh." For one, I think the configuration of the images could be handled better if they were kept separately, but it can be easy to mess up YAML and JSON, so I propose using DHall to maintain the Docker configurations. Next I don't want to build images that already exist, so I've written a Python script that can use the "docker" module to find the tags of existing images and skip building those. We can, of course, have a --force option to make them all.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by robkooper** [comment]
Not sure what did adds. The images for Basgra and BioCro do need to be build since it adds the model_info.json to /work folder

Other problem, this add python and DHall as requirement

Also in general it is best to add the requirements.txt and use virtual environment in python.

**Comment by julianpistorius** [comment]
@robkooper you said:

> The images for Basgra and BioCro do need to be build since it adds the model_info.json to /work folder

Can this be solved by bind-mounting a local, modified version of `model_info.json` at runtime? At least for development purposes?

**Comment by robkooper** [comment]
yup, that is possible. depends also what you are trying to do. Check out development guide pull request as well #2610

File: base/logger/DESCRIPTION [hunk 1]
```diff
@@ -11,5 +11,5 @@ Suggests: testthat
 License: BSD_3_clause + file LICENSE
 Encoding: UTF-8
 LazyData: true
-RoxygenNote: 7.0.2
+RoxygenNote: 7.1.0
```
Comment (infotroph) [inline base/logger/DESCRIPTION:14]: Please keep Roxygen at version 7.0.2 for now -- I know it's annoying to keep the version behind on a dev machine, but when we bump the version we want to do it in all packages at once.

**Comment by infotroph** [comment]
Closing as out of scope—we’d like to avoid adding more dependencies and have mostly addressed the build issue through other means.

**Pull request closed without merging**